packages/autocomplete/Autocomplete.svelte

```diff
@@ -0,0 +1,309 @@
+    });
+  }
+
+  $: activeOptions = activeOptions.filter((el) => el);
+
+  export function blur() {
+    if (anchorElement) {
+      const inputEl = anchorElement.querySelector(
+        "input.mdc-text-field__input"
+      );
+
+      menu && menu.setOpen(false);
+      inputEl && inputEl.blur();
+    }
+  }
+
+  export function selectOption(option, shouldBlur = false) {
+    value = option;
+    formattedValue = getOptionLabel(option);
+    shouldBlur && blur();
+    dispatch("optionselected", option);
+  }
+
+  export function deselectOption(option, shouldBlur = false) {
+    value = undefined;
+    formattedValue = "";
+    shouldBlur && blur();
+    dispatch("optiondeselected", option);
+  }
+
+  export function toggleOption(option, shouldBlur = false) {
+    if (getOptionSelected(option, value)) {
+      deselectOption(option, shouldBlur);
+    } else {
+      selectOption(option, shouldBlur);
+    }
+  }
+
+  function getActiveMenuItems() {
+    const menuItems = menu ? menu.getItems() : undefined;
+    if (menuItems) {
+      return menuItems.filter((el) => !hasClass(el, "mdc-list-item--disabled"));
+    }
+  }
+
+  function handleTextfieldKeydown(e) {
+    if (e.keyCode == 40) {
+      e.preventDefault();
+
+      menu.setOpen(true);
+      if (focusedIndex == -1 || focusedIndex == activeItems.length) {
+        focusedIndex = 0;
+      } else {
+        focusedIndex++;
+      }
+    } else if (e.keyCode == 38) {
+      e.preventDefault();
+
+      menu.setOpen(true);
+      if (focusedIndex == -1 || focusedIndex == 0) {
+        focusedIndex = activeItems.length;
+      } else {
+        focusedIndex--;
+      }
+    } else if (e.keyCode == 27 && clearOnEscape) {
+      e.preventDefault();
+
+      // Clear on escape
+      value = undefined;
+      formattedValue = "";
+      focusedIndex = -1;
+    } else if (e.keyCode == 13) {
+      e.preventDefault();
+
+      if (focusedItem) {
+        activeOptions[focusedIndex].action(e);
+        focusedItem = undefined;
+        focusedIndex = -1;
+      }
+    }
+  }
+
+  function handleTextfieldBlur(e) {
+    focusedIndex = -1;
+    menu && menu.setOpen(false);
+
+    if (clearOnBlur && !value && shouldClearOnBlur) {
+      formattedValue = "";
+    }
+  }
+
+  function hasClass(element, className) {
+    return ` ${element.className} `.indexOf(` ${className} `) > -1;
+  }
+
+  function addClass(element, className) {
+    const arr = element.className.split(" ");
+    if (arr.indexOf(className) == -1) {
+      element.className += " " + className;
+    }
+  }
+
+  function removeClass(element, className) {
+    element.className = element.className.replace(className, "");
+  }
+</script>
+
+<style>
+  * :global(.smui-autocomplete-menu) {
+    width: 100%;
+  }
+</style>
+
+<div use:Anchor bind:this={anchorElement}>
+  <div
+    on:focus|capture={() => {
+      if (!value && menu) {
+        menu.setOpen(true);
+      }
+    }}
+    on:blur|capture={handleTextfieldBlur}>
+    <Textfield
```

Comment:
We could potentially decouple the `Autocomplete` from the `Textfield` by passing as a prop the `Textfield` element to whom the `Autocomplete` should attach. Why? We would profit from this. For instance, by making the Texfield styles decoupled from the `Autocomplete`, the componetns that use the `Texfield` could style it regardless of the auto complete integration, as a normal Textfield.

Reply:
We could pass the `Textfield` as a prop, but then we could have trouble detecting the `blur` event that the `Autocomplete` listens for clearing the `Textfield` if no value was selected. Maybe we could pass it in the default `slot`?

Reply:
A `slot` should serve fine as well 👍 I think that would be awesome

Reply:
Done! 😄